resources/idlharness.js

```diff
@@ -3394,16 +3407,85 @@ function idl_test(srcs, deps, idl_setup_func) {
     }, 'idl_test setup');
 }
 
+// TODO: it would be nice to support `idl_array.add_objects`
 /**
- * fetch_spec is a shorthand for a Promise that fetches the spec's content.
+ * idl_test_shadowrealm is a promise_test wrapper that handles the fetching of the IDL, and
+ * running the code in a `ShadowRealm`, avoiding repetitive boilerplate.
+ *
+ * @see https://github.com/tc39/proposal-shadowrealm
+ * @param {String[]} srcs Spec name(s) for source idl files (fetched from
+ *      /interfaces/{name}.idl).
+ * @param {String[]} deps Spec name(s) for dependency idl files (fetched
+ *      from /interfaces/{name}.idl). Order is important - dependencies from
+ *      each source will only be included if they're already know to be a
+ *      dependency (i.e. have already been seen).
  */
-function fetch_spec(spec) {
-    var url = '/interfaces/' + spec + '.idl';
+function idl_test_shadowrealm(srcs, deps) {
+    const script_urls = [
+        "/resources/testharness.js",
+        "/resources/WebIDLParser.js",
+        "/resources/idlharness.js",
```

Comment:
I'm not a fan of the layering here, where idlharness.js loads itself with dependencies into that realm. And the fetching of IDL happens in the outer idlharness.js but the inner one runs the tests.

What will other tests using `ShadowRealm` look like? Will .any.js support `ShadowRealm` as well?

Can you explore having idlharness.js only once, and doing the fetching inside the `ShadowRealm`?

Reply:
```
<Ms2ger> foolip: so, the thing is that code inside a ShadowRealm doesn't get to fetch anything - only import JS modules
<Ms2ger> foolip: alternatively, I can extract my new code from idlharness.js into its own file, and just duplicate the fetch_spec thing
<foolip> Hmm, I thought I saw fetch() in https://github.com/tc39/proposal-shadowrealm/issues/331, but I didn't.
<Ms2ger> That would have made my life much easier, yes :)
<foolip> So, I can't spend more time in the shadows today, but am willing to be pinged about this again in the future. I always have a lot of time in the future.
<Ms2ger> Heh
<Ms2ger> foolip: in terms of future - this is blocking merging the spec PRs, at least for whatwg. I don't mind holding off a bit, but not long
<foolip> I hope this isn't the only test for ShadowRealm in WPT?
<Ms2ger> foolip: wpt only has the ones for Web APIs; the rest is in https://github.com/tc39/test262/tree/main/test/built-ins/ShadowRealm
```

I pushed the alternative approach; please take a look if you have a moment.

I don't have clear insight in what other tests will look like - at this point I'm not convinced we'll end up writing so many tests that more infrastructure will be needed. I think we should look into that when we have some more tests to extract use cases from.